garak/generators/huggingface.py

```diff
@@ -87,6 +90,10 @@ def _load_client(self):
             self.generator.tokenizer = AutoTokenizer.from_pretrained(
                 pipeline_kwargs["model"]
             )
+        self.use_chat = (
+            hasattr(self.generator.tokenizer, "chat_template")
+            and self.generator.tokenizer.chat_template is not None
+        )
         if not hasattr(self, "deprefix_prompt"):
             self.deprefix_prompt = self.name in models_to_deprefix
```

Comment:
Your read is the intent. Currently I am treating the options as independent. `use_chat` is about expectations related to trained behavior supported by the model for input and output format, and `deprefix` is about expected behavior for the model response format. For this reason and since we `strictly` remove the each value only if it is matched exactly at the beginning of the intermediary output string this seems a reasonable treatment.